app/models/user.rb

```diff
@@ -2,6 +2,7 @@ class User < ApplicationRecord
   has_many :transactions, dependent: :destroy
   has_many :authorizations, dependent: :destroy
   has_many :categories, dependent: :destroy
+  after_create :create_default_categories
```

Comment:
@marderer let's not use `after_...` and `before_...` hooks for models, because of implicity. See [post](https://medium.com/planet-arkency/the-biggest-rails-code-smell-you-should-avoid-to-keep-your-app-healthy-a61fd75ab2d3)

Instead of hooks we can add `Users::Creator` service for creating users: it will be useful because of two reasons:
1) all stuff that we need for user creating is in one place
2) we can reuse this service, even from a console, for example

Reply:
@marderer so, here is what I see as a fast solution:
```ruby
class Users::Creator < Struct.new(:email)
  def call
    create_user
    add_default_categories
  end

  private

  def create_user
    @user = User.create(
      email: email,
      password: generate_password,
      confirmed_at: Date.current
    )
  end

  def generate_password
    Devise.friendly_token[0, 20]
  end

  def add_default_categories
    Categories::Creator.new(@user).create_default_categories
  end
end
```
and in `Users::Omniauth`:
```ruby
def user
  @_user ||= User.find_by(email: email) || create_user
end

def create_user
  Users::Creator.new(email).call
end
```

Reply:
@AleksSenkou it's fine, but service for creating a user returns category list. I added `@user` to `#add_default_category` method. Is it good or better add `return @user` to #call method?